src/social-network-provider/twitter.com/ui/fetch.ts

```diff
@@ -0,0 +1,131 @@
+import {
+    bioCard,
+    fromPostSelectorsSelectPostContentString,
+    postPopupSelector,
+    postsContentSelectors,
+    postsRootSelector,
+    postsSelectors,
+    selfInfoSelectors,
+} from '../utils/selector'
+import { MutationObserverWatcher } from '@holoflows/kit'
+import Services from '../../../extension/service'
+import { PersonIdentifier } from '../../../database/type'
+import { host } from '../index'
+import { getEmptyPostInfo, SocialNetworkUI } from '../../../social-network/ui'
+import { deconstructPayload } from '../../../utils/type-transform/Payload'
+import { regexMatch, timeout } from '../../../utils/utils'
+import { hasPostPopup } from '../utils/status'
+
+export const resolveLastRecognizedIdentity = (self: SocialNetworkUI) => {
+    const selfSelector = selfInfoSelectors.screenName
+    const assign = () => {
+        const ref = self.lastRecognizedIdentity
+        const i = new PersonIdentifier(host, selfInfoSelectors.screenName.evaluateOnce()[0])
+        if (i.isUnknown) throw new Error('failed to recognize user id')
+        ref.value = { identifier: i }
+    }
+    new MutationObserverWatcher(selfSelector)
+        .enableSingleMode()
+        .addListener('onAdd', () => assign())
+        .addListener('onChange', () => assign())
+        .startWatch()
+        .then()
+}
+
+export const resolveInfoFromBioCard = () => {
+    const userAvatarUrl = bioCard
+        .nth(0)
+        .querySelector<HTMLImageElement>('img')
+        .evaluateOnce()[0].src
+    const userNames = bioCard
+        .nth(1)
+        .evaluateOnce()[0]
+        .innerText.split('\n')
+    const userBio = bioCard.nth(2).evaluateOnce()[0].innerText
```

Comment:
You should call `.clone()` before use a shared LiveSelector. LiveSelector has internal state, and any function chains excepts `evaluateOnce` will leads to the changes of the internal state.

And `evaluateOnce` is deprecated, use `evaluate` instead

Reply:
TS2339: Property 'evaluate' does not exist on type 'LiveSelector<HTMLImageElement, true>'.

Reply:
Ok your package from github is not auto upgrading. ```yarn upgrade``` required. Now a ```just``` script will do this at post install time. We will need npm managing version of this package ASAP.

Reply:
8ec9f6c5

Reply:
f1882da7. Now always get new selector. check it out.